scripts/asr_language_modeling/ngram_lm/eval_beamsearch_ngram.py

```diff
@@ -13,71 +13,139 @@
 # limitations under the License.
 #
 
+"""
 # This script would evaluate an N-gram language model trained with KenLM library (https://github.com/kpu/kenlm) in
 # fusion with beam search decoders on top of a trained ASR model. NeMo's beam search decoders are capable of using the
 # KenLM's N-gram models to find the best candidates. This script supports both character level and BPE level
 # encodings and models which is detected automatically from the type of the model.
 # You may train the LM model with 'scripts/ngram_lm/train_kenlm.py'.
-#
-# USAGE: python eval_beamsearch_ngram.py --nemo_model_file <path to the .nemo file of the model> \
-#                                         --input_manifest <path to the evaluation JSON manifest file \
-#                                         --kenlm_model_file <path to the binary KenLM model> \
-#                                         --beam_width <list of the beam widths> \
-#                                         --beam_alpha <list of the beam alphas> \
-#                                         --beam_beta <list of the beam betas> \
-#                                         --preds_output_folder <optional folder to store the predictions> \
-#                                         --decoding_mode beamsearch_ngram
-#                                         ...
-#
+
+# Config Help
+
+To discover all arguments of the script, please run :
+python eval_beamsearch_ngram.py --help
+python eval_beamsearch_ngram.py --cfg job
+
+# USAGE
+
+python eval_beamsearch_ngram.py nemo_model_file=<path to the .nemo file of the model> \
+           input_manifest=<path to the evaluation JSON manifest file \
+           kenlm_model_file=<path to the binary KenLM model> \
+           beam_width=[<list of the beam widths, separated with commas>] \
+           beam_alpha=[<list of the beam alphas, separated with commas>] \
+           beam_beta=[<list of the beam betas, separated with commas>] \
+           preds_output_folder=<optional folder to store the predictions> \
+           probs_cache_file=null \
+           decoding_mode=beamsearch_ngram
+           ...
+
+
+# Grid Search for Hyper parameters
+
+For grid search, you can provide a list of arguments as follows -
+
+           beam_width=[4,8,16,....] \
```

Comment:
In many cases, user just want to try one combination, does the code support single integer instead of list for beam width, alpha and beta? If not, how about having such a feature?

Reply:
What happens to hydra when an integer is passes? raises an error?
If not, can we just check the type and convert it to a list at the beginning of the script?